mujoco_warp/_src/smooth.py

```diff
@@ -165,24 +165,22 @@ def _geom_local_to_global(
   # Data in:
   xpos_in: wp.array2d(dtype=wp.vec3),
   xquat_in: wp.array2d(dtype=wp.quat),
-  geom_skip_in: wp.array(dtype=bool),
   # Data out:
   geom_xpos_out: wp.array2d(dtype=wp.vec3),
   geom_xmat_out: wp.array2d(dtype=wp.mat33),
-  geom_skip_out: wp.array(dtype=bool),
 ):
   worldid, geomid = wp.tid()
   bodyid = geom_bodyid[geomid]
-  if not geom_skip_in[geomid]:
-    # Calculate only if necessary
-    xpos = xpos_in[worldid, bodyid]
-    xquat = xquat_in[worldid, bodyid]
-    geom_xpos_out[worldid, geomid] = xpos + math.rot_vec_quat(geom_pos[worldid % geom_pos.shape[0], geomid], xquat)
-    geom_xmat_out[worldid, geomid] = math.quat_to_mat(math.mul_quat(xquat, geom_quat[worldid % geom_quat.shape[0], geomid]))
 
-    if bodyid == 0:
-      # static geom pose are calculated only once
-      geom_skip_out[geomid] = True
+  if bodyid == 0:
```

Comment:
This is not comprehensive, ya? What we want is `body_weldid[bodyid] == 0` which will also cover geoms in static bodies that are descendants of the world, not just direct world geoms.

This is not hypothetical hair splitting either. E.g. @kevinzakka, for reasons that I do not condone but are also not illegitimate, puts all his very fancy terrains in a static body under the world, which this optimization will miss.